### ISSUE TITLE
fp-SVG's plot() ignores the last past frame + a fix suggestion

### ISSUE THREAD

**Issue by badooki** [edenton/svg#11]
Hi,

The plot() function in train_svg_fp.py has an issue of disregarding the last frame of the past frames, before moving on to predicting future frames.

At the end of `if i < opt.n_past:` block (<`for i in range(1, opt.n_eval):`<`for s in range(nsample):`<`def plot(x, epoch):`) (line 189), I suggest adding the following lines:

`h, skip= h_seq[i]`
`h = h.detach()`

With this change the following will happen:
`h` will be overwritten by `h, skip = h_seq[i-1]` in line 180, until `i` < opt.n_past. However, when the loop updates/iterates `i` from `opt.n_past - 1` to `opt.n_past`, `h` and `skip` from the last of the past frames will be used in the `else:` block starting in line 190.

lp-SVG code does not suffer from this issue, but skip connection still comes from a second last frame of the past frames, which I do not think is a big issue (although may not be ideal).
